server/game/cards/03_TWI/units/AsajjVentressCountDookusAssassin.ts

```diff
@@ -0,0 +1,35 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { Trait } from '../../../core/Constants';
+import { StateWatcherRegistrar } from '../../../core/stateWatcher/StateWatcherRegistrar';
+import { AttacksThisPhaseWatcher } from '../../../stateWatchers/AttacksThisPhaseWatcher';
+
+export default class AsajjVentressCountDookusAssassin extends NonLeaderUnitCard {
+    private attacksThisPhaseWatcher: AttacksThisPhaseWatcher;
+
+    protected override getImplementationId () {
+        return {
+            id: '3556557330',
+            internalName: 'asajj-ventress#count-dookus-assassin',
+        };
+    }
+
+    protected override setupStateWatchers(registrar: StateWatcherRegistrar) {
+        this.attacksThisPhaseWatcher = AbilityHelper.stateWatchers.attacksThisPhase(registrar, this);
+    }
+
+    public override setupCardAbilities () {
+        this.addOnAttackAbility({
+            title: 'If you\'ve attacked with another Separatist unit this phase, this unit gets +3/+0 for this phase.',
+            immediateEffect: AbilityHelper.immediateEffects.conditional({
+                condition: (context) => this.attacksThisPhaseWatcher.getAttackersInPlay((attack) => attack.attacker.hasSomeTrait(Trait.Separatist) && attack.attacker !== context.source).length > 0,
```

Comment:
This isn't quite right, due to the SWU rules around the idea of a "copy" of a card - i.e., every time a card enters play, it becomes a new "copy" for history purposes. So e.g. if Asajj attacks, is defeated, then played again, she would in that case count her previous attack as "another unit" doing a Separatist attack since she is now a new copy.

The way to implement that correctly should be:

```ts
  condition: (context) => this.attacksThisPhaseWatcher.someUnitAttackedControlledByPlayer({
      controller: context.source.controller,
      filter: (attackEvent) =>
          attackEvent.attacker.hasSomeTrait(Trait.Separatist) &&
          (attackEvent.attacker !== context.source || attackEvent.attackerInPlayId !== context.source.inPlayId)
  }),
```

Also we need to add a test confirming the case works, but I left a separate comment for that

Reply:
Thanks for explaining clearly this rule point. I'll try to implement it (not sure yet how to achieve it).

Reply:
I do not understand why we we need this `attackEvent.attacker !== context.source`?
Is it because before the end of the first attack, there is no `attackerInPlayId`?

Reply:
The inPlayId is just an incrementing counter that's unique to each card. So we can't compare them across two different card objects, only repeated plays of the same card.

Another way to think about it is that the statement `attackEvent.attacker !== context.source || attackEvent.attackerInPlayId !== context.source.inPlayId` is basically saying that a card counts as being a "different" card if either it's a different card than this one altogether; OR if it's the same card, it was a previous instance of it (previous inPlayId)